src/primval.rs

```diff
@@ -14,6 +14,8 @@ pub enum PrimVal {
 
     Ptr(Pointer),
     FnPtr(Pointer),
+    VtablePtr(Pointer, Pointer),
+    SlicePtr(Pointer, u64),
```

Comment:
The alternative would require some panicking expect-methods to extract the extra data, since it would allow any primvals in extra and pointer position